token-service/src/main/java/org/p2p/token/service/manager/TokenServiceEventPublisher.kt

```diff
@@ -0,0 +1,65 @@
+package org.p2p.token.service.manager
+
+import kotlin.coroutines.CoroutineContext
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.flow.distinctUntilChanged
+import kotlinx.coroutines.flow.filterNot
+import kotlinx.coroutines.launch
+import org.p2p.core.dispatchers.CoroutineDispatchers
+import org.p2p.core.token.SolAddress
+import org.p2p.core.wrapper.eth.EthAddress
+import org.p2p.token.service.interactor.TokenServiceInteractor
+import org.p2p.token.service.model.TokenServiceNetwork
+
+internal class TokenServiceEventPublisher(
+    private val tokenServiceInteractor: TokenServiceInteractor,
+    private val coroutineDispatcher: CoroutineDispatchers,
+    private val eventManager: TokenServiceEventManager
+) : CoroutineScope {
+    override val coroutineContext: CoroutineContext = coroutineDispatcher.io
+
+    init {
+        launch { attachToTokensPriceFlow(TokenServiceNetwork.SOLANA) }
+    }
+
+    fun loadTokensPrice(networkChain: TokenServiceNetwork, addresses: List<String>) {
+        launch {
+            eventManager.notify(
+                eventType = TokenServiceEventType.from(networkChain),
+                event = TokenServiceEvent.Loading
```

Comment:
you can remove new entity like event and just observe to some sealed class with values Loading / Content / Error